aten/src/ATen/native/native_functions.yaml

```diff
@@ -14486,6 +14486,11 @@
     NestedTensorCUDA: _scaled_dot_product_flash_attention_nestedtensor_cuda
   tags: nondeterministic_seeded
 
+- func: _scaled_dot_product_flash_attention_mask(Tensor query, Tensor key, Tensor value, float dropout_p=0.0, bool is_causal=False, bool return_debug_mask=False, *, Tensor? attn_mask=None, float? scale=None) -> (Tensor output, Tensor logsumexp, Tensor cum_seq_q, Tensor cum_seq_k, SymInt max_q, SymInt max_k, Tensor philox_seed, Tensor philox_offset, Tensor debug_attn_mask)
```

Comment:
Okay so if we have to make a new aten function to support the extended functionality, I think we should just simplify the device dispatching/ meta func stuff. Why not just have a _sdpa_flash_cpu, where you can define the args for exactly what you support and always call this function. This would simplify the meta_registration as well

Reply:
which is partially what you did it seems but I think we should remove one more layer of indirection

Reply:
Sure! I have ceated `_sdpa_flash_cpu` specifically for CPU and removed useless args.